pydantic/_internal/_core_utils.py

```diff
@@ -81,8 +81,11 @@ def get_type_ref(type_: type[Any], args_override: tuple[type[Any], ...] | None =
         args = generic_metadata['args'] or args
 
     module_name = getattr(origin, '__module__', '<No __module__>')
-    qualname = getattr(origin, '__qualname__', f'<No __qualname__: {origin}>')
-    type_ref = f'{module_name}.{qualname}:{id(origin)}'
+    if isinstance(origin, TypeAliasType):
+        type_ref = f'{module_name}.{origin.__name__}'
```

Comment:
No point in having the `id` there for `TypeAliasType` and since this ends up in the JsonSchema that gives us a cleaner name.